From e4de1621c2b619ff4a78f37f5049ee7b35927e80 Mon Sep 17 00:00:00 2001
From: Hien Huynh <hien.huynh.px@renesas.com>
Date: Wed, 23 Dec 2020 09:47:00 +0700
Subject: [PATCH 21/22] rzg: ddr: Update DDR setting for G2H, G2M and G2N.

This commit refers from:
- Commit: <9c1bfc5d4497006ec86dd749c92c9bf3cda3fc8a>
- Branch: rcar_gen3
- Repository: https://github.com/renesas-rcar/arm-trusted-firmware

Reference commit updates H3, M3, M3N DDR setting rev.0.41.
Apply it for rzg platform with G2H, G2M and G2N.

Signed-off-by: Hien Huynh <hien.huynh.px@renesas.com>
---
 drivers/renesas/rzg/ddr/ddr_b/boot_init_dram.c        | 7 ++++++-
 drivers/renesas/rzg/ddr/ddr_b/boot_init_dram_regdef.h | 2 +-
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram.c b/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram.c
index fad5700..629ea0e 100644
--- a/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram.c
+++ b/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram.c
@@ -3587,7 +3587,12 @@ int32_t rzg_dram_init(void)
 	}
 
 	/* THCTR Bit6: PONM=0 , Bit0: THSST=0   */
-	data_l = mmio_read_32(THS1_THCTR) & 0xFFFFFFBE;
+	data_l = mmio_read_32(THS1_THCTR);
+	if (data_l & 0x00000040U)
+		data_l = data_l & 0xFFFFFFBEU;
+	else
+		data_l = data_l | 0x00000001U;
+
 	mmio_write_32(THS1_THCTR, data_l);
 
 	/* Judge product and cut */
diff --git a/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram_regdef.h b/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram_regdef.h
index 565bcbd..2669196 100644
--- a/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram_regdef.h
+++ b/drivers/renesas/rzg/ddr/ddr_b/boot_init_dram_regdef.h
@@ -5,7 +5,7 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-#define RZG_DDR_VERSION		"rev.0.40"
+#define RZG_DDR_VERSION		"rev.0.41"
 #define DRAM_CH_CNT		0x04
 #define SLICE_CNT		0x04
 #define CS_CNT			0x02
-- 
2.7.4

